tests/frontend_tests/TrialTable.test.tsx

```diff
@@ -0,0 +1,135 @@
+import React from "react"
+import Enzyme from "enzyme"
+import Adapter from "@wojtekmaj/enzyme-adapter-react-17"
+import { mount } from "enzyme"
+
+global.URL.createObjectURL = jest.fn()
+
+import { TrialTable } from "../../optuna_dashboard/static/components/StudyDetail"
+import {
+  DataGridColumn,
+  descendingComparator,
+} from "../../optuna_dashboard/static/components/DataGrid"
+
+Enzyme.configure({ adapter: new Adapter() })
+const trials = [
+  {
+    trial_id: 1,
+    study_id: 0,
+    number: 0,
+    state: "Complete" as TrialState,
+    values: [-1],
+    intermediate_values: [],
+    datetime_start: new Date("2021-06-15T00:00:00"),
+    datetime_complete: new Date("2021-06-15T00:00:01"),
+    params: [
+      { name: "x", value: "1" },
+      { name: "y", value: "2" },
+    ],
+    user_attrs: [],
+    system_attrs: [],
+  },
+  {
+    trial_id: 2,
+    study_id: 0,
+    number: 1,
+    state: "Complete" as TrialState,
+    values: [-2],
+    intermediate_values: [],
+    datetime_start: new Date("2021-06-15T00:00:01"),
+    datetime_complete: new Date("2021-06-15T00:00:02"),
+    params: [
+      { name: "x", value: "2" },
+      { name: "y", value: "1" },
+    ],
+    user_attrs: [],
+    system_attrs: [],
+  },
+]
+
+const study_direction: StudyDirection = "minimize" as StudyDirection
+
+const study_detail = {
+  name: "study_0",
+  directions: [study_direction],
+  datetime_start: new Date("2021-06-15T00:00:00"),
+  best_trial: trials[1],
+  trials: trials,
+  intersection_search_space: [
+    {
+      name: "x",
+      type: "UniformDistribution",
+      attributes: { low: -3, high: 3 },
+    },
+    {
+      name: "y",
+      type: "UniformDistribution",
+      attributes: { low: -3, high: 3 },
+    },
+  ],
+  union_search_space: [
+    {
+      name: "x",
+      type: "UniformDistribution",
+      attributes: { low: -3, high: 3 },
+    },
+    {
+      name: "y",
+      type: "UniformDistribution",
+      attributes: { low: -3, high: 3 },
+    },
+  ],
+}
+
+test("Trial table correctly rendered the trial list", () => {
+  const trial_table_component = mount(<TrialTable studyDetail={study_detail} />)
+  const head_cells = trial_table_component
+    .find("ForwardRef(TableHead)")
+    .find("WithStyles(ForwardRef(TableCell))")
+
+  const columns: DataGridColumn<Trial>[] = trial_table_component
+    .find("DataGrid")
+    .prop("columns")
+
+  for (let i = 0; i < head_cells.length; i++) {
```

Comment:
These codes are known as one factor that makes tests more complicated than they really should be. Could you make tests more simple?
http://xunitpatterns.com/Conditional%20Test%20Logic.html